src/config.c

```diff
@@ -1885,10 +1911,15 @@ static void enumConfigInit(standardConfig *config) {
 }
 
 static int enumConfigSet(standardConfig *config, sds *argv, int argc, const char **err) {
```

Comment:
we need to also handle enumConfigGet and also possibly enumConfigRewrite

Reply:
Changes done.
I'm considering removing methods `configEnumGetNameOrUnknown` and `configEnumGetName`, to maintain only `configEnumGetNamesOrUnknown`.
What do you think?

Reply:
not sure if you're serious of kidding, so to be on the safe side i'll state:
configEnumGetNameOrUnknown return a static string, that does not need to be released.

but the new configEnumGetNamesOrUnknown could either return a static string, or a heap allocated one (that should be released, and that fact should be probably mentioned in its doc comment).
this must be resolved (it should always return a heap allocated one (use sdsnew on the static one), and make sure that the caller always release it (it means that enumConfigGet should not pass it though another sdsnew)

Reply:
Thanks. I was not kidding but found out the actual leak was because I wasn't considering we could load unflagged enums.
Changed to always use specialized parser for single vs multiarg, + protect from a leak in case the multiarg config has a unflagged enum.
Not sure we can have a good way of validating the enums.